### ISSUE TITLE
circular imports

### ISSUE THREAD

**Issue by saidctb** [pyccel/sympde#78]
the file `sympde.topology.mapping` and `sympde.topology.domain` have a circular dependecy
the file `sympde.topology.mapping` and `sympde.expr.expr` have a circular dependecy

**Comment by yellalena** [comment]
hey can I work on this?

**Comment by saidctb** [comment]
> hey can I work on this?

Sure it will be great, we can discuss your proposed solution here before starting to code  and see if it's suitable

**Comment by yellalena** [comment]
yes okay give me a day or a few, i'll find some time and figure it out. brb :))

**Comment by yellalena** [comment]
@saidctb hey! i finally managed to find time for this task. could you please tell me how to I run this app locally so that I can see the problem? because all tests pass.

**Comment by saidctb** [comment]
> @saidctb hey! i finally managed to find time for this task. could you please tell me how to I run this app locally so that I can see the problem? because all tests pass.

you just need to run `pytest sympde/` after installing it

**Comment by yellalena** [comment]
yeah but when I run it, all tests pass.
